src/app/content/studyGuides/selectors.spec.ts

```diff
@@ -13,11 +14,12 @@ jest.mock('../selectors', () => ({
 describe('studyGuidesLocationFiltersWithContent', () => {
   it('filters', () => {
     mockBook.mockReturnValue({id: 'enabledbook', tree: treeWithoutUnits});
+    const section = findArchiveTreeNodeById(treeWithoutUnits, 'chapter1')!;
```

Comment:
What is the purpose of using exclamation mark here?

Reply:
`findArchiveTreeNodeById` is returning `LinkedArchiveTreeNode | undefined`
`!` is a typescript feature that exclude `undefined` type so we are saying to the typescript that in this case the `section` is for sure `LinkedArchiveTreeNode`

We are using this only in tests if we know for sure that a statement is true.
In the production code you can see that we are using `assertDefined` helper that will throw an error with a proper message if something is not right